### ISSUE TITLE
2019-12-01 tc015 M3SD FLGACCEPTTMAX is missing from generated parameter results

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#73]
FLGACCEPTTMAX should be generated if requested by MCT$PARAMETERLIST even if FLAGS$FLGEMESIS doesn't exist. With commit 6e99fab it isn't generated for tc015 M3SD. Will need to check other test cases if this is the case.

Also missing from tc016 M3SS, tc017 M4SD

**Comment by rudraya** [comment]
I am noticing that the test cases for M3SD, M3SS and M4SD dont have FLGEMESIS in the MCT. Please make sure that FLGEMESIS is in the MCT (Just how FLEXKEL, FLGEXAUC, FLGEXSDE are present). Given that as a pre-req, I am able to generate FLGACCEPTTMAX.

**Comment by tensfeldt** [comment]
FLGEMESIS is not a defined MCT keyword/element. It's a "possible" dataset value but not required. Review of the Computation Engine Specifications document appears to support that. FLGEMESIS is a time based flag that is intended only to be present in the input concentration dataset itself.

**Comment by rudraya** [comment]
Just want to clarify based on the comment above, would the data represented by FLGEMESIS ever have a different name other than "FLGEMESIS". The only reason I am bring this up is because usually the computation engine checks the MCT keyword/element pair as a reference before looking at the dataset for the column name, and I want to account for a case where it is named differently in the dataset (if that is even a possibility). Based on your comment, I will make the appropriate changes in the next commit.

**Comment by tensfeldt** [comment]
We can assume that there CAN be a definition in the MCT of FLGEMESIS and it will define the name of the column in the dataset. The default will be that mct$FLGEMESIS=="FLGEMESIS".

**Comment by tensfeldt** [comment]
If I add FLGEMESIS to mct with value of "FLGEMESIS" now tc008 M1SD fails to run with the following error message:

```
> mct$FLGEMESIS
[1] "FLGEMESIS"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in `[.data.frame`(tmp_df, , map_data$FLGEMESIS) : 
  undefined columns selected
In addition: Warning messages:
1: In run_M1_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTTAU' cannot be computed if 'TAUi' is not provided
2: In run_M1_SD_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in `[.data.frame`(tmp_df, , map_data$FLGEMESIS) : 
  undefined columns selected 
```
Please let me know when you've adjusted the code to not require a value of FLGEMESIS in the dataset. It should be optional in both MCT and dataset. If it is defined in MCT, the code should attempt to find a column with the name of the value of MCT$FLGEMESIS ("FLGEMESIS" by default in the MCT definitions) but not require that a value must appear in the dataset. I've updated all of the local (to my machine) definitions of the mcts to include FLGEMESIS now with value "FLGEMESIS" but I suspect each and every test case will now fail.

**Comment by rudraya** [comment]
This issue was addressed in bb8e82d, please let me know if it still occurs.

**Comment by tensfeldt** [comment]
Retested with commit bb8e82d and tc015. FLGACCEPTTMAX is now generated for the output parameter results and presence of FLGEMESIS in MCT no longer causes crash.
This issue will be closed but FLGEMESIS in MCT will be tested with other test cases.
